src/openassetio-core-c/tests/hostApi/ManagerTest.cpp

```diff
@@ -187,7 +187,7 @@ SCENARIO("A host calls Manager::identifier") {
     // Create mock ManagerInterface to inject and assert on.
     const managerApi::ManagerInterfacePtr mockManagerInterfacePtr =
         std::make_shared<MockManagerInterface>();
-    auto& mockManagerInterface = static_cast<MockManagerInterface&>(*mockManagerInterfacePtr);
+    auto& mockManagerInterface = dynamic_cast<MockManagerInterface&>(*mockManagerInterfacePtr);
```

Comment:
Just calling out I didn't see casts (particularly dynamic_casts) appear in the description, so checking if this needs some words to explain to readers?

Reply:
Yeah there are a couple of changes to satisfy cppcoreguidelines that I didn't put in the description (e.g. uninitialised variables), since I thought they didn't really need further explanation. Do you think this change is controversial enough to warrant flagging explicitly in the description/commit?

Reply:
`dynamic_cast` tends to catch my eyes as to why is this needed. Unless all the consumers are well aware of this? Maybe not an issue as the library is compiled with RTTI, but that tends to be a sticky point IMO.

Reply:
I think its reasonable that casting from base to derived should be a `dynamic_cast` for safety, as a general rule. 

In this case we can be certain of the provenance of the object, so perhaps `static_cast` isn't so bad after all? 

This does only affect unit tests so is not so important.

Indeed OpenAssetIO will have all sorts of problems without RTTI.

Otherwise, I'll add a note about it in the commit message when squashing down the commits.

Reply:
Ahh there you go, hadn't realised it was intentionally from base to derived. All good!